api/main.py

```diff
@@ -39,11 +39,19 @@
 
 
 @app.exception_handler(RequestValidationError)
-async def handle_req_validation_error(req: Request, exc: RequestValidationError) -> JSONResponse:
+async def handle_req_validation_error(
+    req: Request, exc: RequestValidationError
+) -> JSONResponse:
     """A default handler to handle malformed request bodies."""
     return JSONResponse(status_code=400, content={"error": exc.errors()})
 
 
+@app.exception_handler(ValueError)
+async def handle_validation_value_error(req: Request, exc: ValueError) -> JSONResponse:
+    """A default handler to handle value errors raised by pydantic model field validators."""
+    return JSONResponse(status_code=400, content={"error": f"{exc}"})
+
+
```

Comment:
That didn't work for me 🤷🏻 though according to me it should have.

Reply:
It worked for me, and according to the [documentation](https://fastapi.tiangolo.com/tutorial/handling-errors/#override-request-validation-exceptions) that is the preferred way.
(I believe the problem might have been with the `404` instead of `400`.
If not, what was the test payload that failed to `error out`?